Warnings with clang 3.9

```
clang++ -ggdb -fno-elide-constructors -fPIC -DPIC -pthread -std=c++14 -I.. -Wall -Wextra -pedantic -Wundef -Wshadow -Wunreachable-code -Wshorten-64-to-32 -Wold-style-cast -Wconditional-uninitialized -Wextra-semi -Wpessimizing-move -DREALM_HAVE_CONFIG -DREALM_DEBUG -MMD -MP -c group_shared.cpp -o group_shared.dbg.pic.o
clang++ -ggdb -fno-elide-constructors -fPIC -DPIC -pthread -std=c++14 -I.. -Wall -Wextra -pedantic -Wundef -Wshadow -Wunreachable-code -Wshorten-64-to-32 -Wold-style-cast -Wconditional-uninitialized -Wextra-semi -Wpessimizing-move -DREALM_HAVE_CONFIG -DREALM_DEBUG -MMD -MP -c group_writer.cpp -o group_writer.dbg.pic.o
clang++ -ggdb -fno-elide-constructors -fPIC -DPIC -pthread -std=c++14 -I.. -Wall -Wextra -pedantic -Wundef -Wshadow -Wunreachable-code -Wshorten-64-to-32 -Wold-style-cast -Wconditional-uninitialized -Wextra-semi -Wpessimizing-move -DREALM_HAVE_CONFIG -DREALM_DEBUG -MMD -MP -c impl/continuous_transactions_history.cpp -o impl/continuous_transactions_history.dbg.pic.o
group_shared.cpp:262:30: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
            data[i].next = i + 1;
                         ~ ~~^~~
group_shared.cpp:266:19: warning: implicit conversion loses integer precision: 'uint_fast32_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        entries = new_entries;
                ~ ^~~~~~~~~~~
group_shared.cpp:344:23: warning: implicit conversion loses integer precision: 'uint_fast32_t' (aka 'unsigned long') to '__int_type' (aka 'unsigned int') [-Wshorten-64-to-32]
        put_pos.store(next(), std::memory_order_release);
                ~~~~~ ^~~~~~
```

--- a/src/realm/group_shared.cpp
+++ b/src/realm/group_shared.cpp
@@ -214,7 +214,7 @@ public:
         // Release is triggered by explicitly storing into count whenever a
         // new entry has been initialized.
         mutable std::atomic<uint32_t> count;
-        uint32_t next;
+        uint_fast32_t next;
     };
 
     Ringbuffer() noexcept
@@ -360,9 +360,9 @@ public:
 
 private:
     // number of entries. Access synchronized through put_pos.
-    uint32_t entries;
-    std::atomic<uint32_t> put_pos; // only changed under lock, but accessed outside lock
-    std::atomic<uint32_t> old_pos; // only changed during write transactions and under lock
+    uint_fast32_t entries;
+    std::atomic<uint_fast32_t> put_pos; // only changed under lock, but accessed outside lock
+    std::atomic<uint_fast32_t> old_pos; // only changed during write transactions and under lock
 
     const static int init_readers_size = 32;
 

